From b71acedbdfd59d8e4b7c3a3b797867535f8de442 Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Fri, 28 Jul 2017 15:37:57 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399: add cpu pvtm voltage table

stress test:
1. reboot
2. antutu, use governor performance
3. antutu, use governor interactive
4. Thomas-sRoomIII, use governor interactive
5. Thomas-sRoomIII, use governor userspace and sweep frequency

Change-Id: If12d2bd72ce3bba01021314265eba4f83a0072e1
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi | 88 ++++++++++++++++++++
 1 file changed, 88 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
index 366d7e70e3c4..0e58ae354664 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
@@ -50,35 +50,75 @@
 		nvmem-cells = <&cpul_leakage>;
 		nvmem-cell-names = "cpu_leakage";
 
+		rockchip,pvtm-voltage-sel = <
+			0        143500   0
+			143501   148500   1
+			148501   152000   2
+			152001   999999   3
+		>;
+		rockchip,pvtm-freq = <408000>;
+		rockchip,pvtm-volt = <1000000>;
+		rockchip,pvtm-ch = <0 0>;
+		rockchip,pvtm-sample-time = <1000>;
+		rockchip,pvtm-number = <10>;
+		rockchip,pvtm-error = <1000>;
+		rockchip,pvtm-ref-temp = <41>;
+		rockchip,pvtm-temp-prop = <115 66>;
+		rockchip,pvtm-thermal-zone = "soc-thermal";
+
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
 			opp-microvolt = <800000>;
+			opp-microvolt-L0 = <800000>;
+			opp-microvolt-L1 = <800000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <800000>;
+			opp-microvolt-L0 = <800000>;
+			opp-microvolt-L1 = <800000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
 			opp-microvolt = <850000>;
+			opp-microvolt-L0 = <850000>;
+			opp-microvolt-L1 = <825000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
 			opp-microvolt = <925000>;
+			opp-microvolt-L0 = <925000>;
+			opp-microvolt-L1 = <900000>;
+			opp-microvolt-L2 = <875000>;
+			opp-microvolt-L3 = <850000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
 			opp-microvolt = <1000000>;
+			opp-microvolt-L0 = <1000000>;
+			opp-microvolt-L1 = <975000>;
+			opp-microvolt-L2 = <950000>;
+			opp-microvolt-L3 = <925000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1416000000 {
 			opp-hz = /bits/ 64 <1416000000>;
 			opp-microvolt = <1125000>;
+			opp-microvolt-L0 = <1125000>;
+			opp-microvolt-L1 = <1100000>;
+			opp-microvolt-L2 = <1075000>;
+			opp-microvolt-L3 = <1050000>;
 			clock-latency-ns = <40000>;
 		};
 	};
@@ -90,45 +130,93 @@
 		nvmem-cells = <&cpub_leakage>;
 		nvmem-cell-names = "cpu_leakage";
 
+		rockchip,pvtm-voltage-sel = <
+			0        149000   0
+			149001   155000   1
+			155001   160000   2
+			160001   999999   3
+		>;
+		rockchip,pvtm-freq = <408000>;
+		rockchip,pvtm-volt = <1000000>;
+		rockchip,pvtm-ch = <1 0>;
+		rockchip,pvtm-sample-time = <1000>;
+		rockchip,pvtm-number = <10>;
+		rockchip,pvtm-error = <1000>;
+		rockchip,pvtm-ref-temp = <41>;
+		rockchip,pvtm-temp-prop = <71 35>;
+		rockchip,pvtm-thermal-zone = "soc-thermal";
+
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
 			opp-microvolt = <800000>;
+			opp-microvolt-L0 = <800000>;
+			opp-microvolt-L1 = <800000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <800000>;
+			opp-microvolt-L0 = <800000>;
+			opp-microvolt-L1 = <800000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
 			opp-microvolt = <825000>;
+			opp-microvolt-L0 = <825000>;
+			opp-microvolt-L1 = <825000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
 			opp-microvolt = <875000>;
+			opp-microvolt-L0 = <875000>;
+			opp-microvolt-L1 = <850000>;
+			opp-microvolt-L2 = <850000>;
+			opp-microvolt-L3 = <850000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
 			opp-microvolt = <950000>;
+			opp-microvolt-L0 = <950000>;
+			opp-microvolt-L1 = <925000>;
+			opp-microvolt-L2 = <900000>;
+			opp-microvolt-L3 = <875000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1416000000 {
 			opp-hz = /bits/ 64 <1416000000>;
 			opp-microvolt = <1025000>;
+			opp-microvolt-L0 = <1025000>;
+			opp-microvolt-L1 = <1000000>;
+			opp-microvolt-L2 = <1000000>;
+			opp-microvolt-L3 = <975000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1608000000 {
 			opp-hz = /bits/ 64 <1608000000>;
 			opp-microvolt = <1100000>;
+			opp-microvolt-L0 = <1100000>;
+			opp-microvolt-L1 = <1075000>;
+			opp-microvolt-L2 = <1050000>;
+			opp-microvolt-L3 = <1025000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1800000000 {
 			opp-hz = /bits/ 64 <1800000000>;
 			opp-microvolt = <1200000>;
+			opp-microvolt-L0 = <1200000>;
+			opp-microvolt-L1 = <1175000>;
+			opp-microvolt-L2 = <1150000>;
+			opp-microvolt-L3 = <1125000>;
 			clock-latency-ns = <40000>;
 		};
 	};
-- 
2.35.3

